postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py

```diff
@@ -0,0 +1,387 @@
+    
+parser.add_argument('--rotate',  dest='rotate', action='store_true' , default = False, 
+                    help='horizontal components are rotated w.r.t. the slip centroid') 
+                    # -fault.xdmf file needed
+    
+parser.add_argument('--outputprefix', default="data", type=str, metavar=('outputprefix'), 
+                    help='provide an outputprefix')
+
+parser.add_argument('--faultxdmf', type=str, metavar=('-fault.xdmf'), 
+                    help='provide path+filename-fault.xdmf; only needed when the path differs from the -surface.xdmf file')
+
+parser.add_argument('--parallelLoading',  dest='parallelLoading', action='store_true' , default = False, 
+                    help='load components in parallel (weak scaling)')
+                    # This can lead to an enconding error of multiprocessing, when the input arrays are too large
+
+parser.add_argument('--avgSWaveVelocity', nargs=1, metavar=('avgSWaveVelocity'),
+                    help='provide an average S-wave velocity near the surface (otherwise it is approximated from P-arrivals)',
+                    type=float)
+
+parser.add_argument('--oldVelocityVariables', dest='oldVelocityVariables', action='store_true' , default = False, 
+                    help='for data with SeisSols old surface velocity variables name convention (uvw)') 
+args = parser.parse_args()
+
+def LoadSingleComponentParallel(component, surfaceData=True):
+    chunks = np.int_(np.linspace(0, nElements, nprocs+1)) if surfaceData else np.int_(np.linspace(0, nElementsFault, nprocs+1))
+    chunks = np.array([chunks[:-1], chunks[1:]-chunks[:-1], np.full(nprocs, component)]).T
+    chunks = [tuple(i) for i in chunks]
+    
+    if surfaceData:
+        p = Pool(nprocs)
+        result = p.map(ReadSurfaceWithPool, chunks)
+        p.close()
+        p.join()
+        return np.concatenate(result, axis=1).T[::stepsize]
+    else:
+        p = Pool(nprocs)
+        result = p.map(ReadFaultWithPool, chunks)
+        p.close()
+        p.join()
+        return np.concatenate(result, axis=1).T[::stepsize]
+    
+def ReadSurfaceWithPool(chunk):
+    return surfacexdmf.ReadDataChunk(chunk[2], int(chunk[0]), int(chunk[1]))[timeIndices[0]:timeIndices[1]]
+
+def ReadFaultWithPool(chunk):
+    return faultxdmf.ReadDataChunk(chunk[2], int(chunk[0]), int(chunk[1]))[timeIndicesFault[0]:timeIndicesFault[1]]
+
+def ComputeTriangleMidpoints(geom, connect):
+    """Generates an array with coordinates of triangle midpoints (same dimension as connect)"""
+    xyz = np.zeros_like(connect, dtype=float)
+    xyz = (1./3.)*(geom[connect[:,0],:]+geom[connect[:,1],:]+geom[connect[:,2],:])   
+    return xyz
+
+def CalculateSlipCentroid(faultxdmf, events=0):
+    ASl = faultxdmf.ReadData("ASl", idt=timeIndicesFault[1]-1).T
+    if timeIndicesFault[0]!=0:
+        ASl -= faultxdmf.ReadData("ASl", idt=timeIndicesFault[0]).T
+    return np.average(faultxyz, axis=0, weights=ASl)
```

Comment:
faultxyz could be passed as argument to the function call (thus avoid global variable)

Reply:
It is also needed in ApproximateEventDurationAndHypocenter()

Reply:
ok then add it also in the call of ApproximateEventDurationAndHypocenter?

Reply:
Ok, done.

Reply:
ha ok, maybe there was a misunderstanding, I was only suggesting 
```
def CalculateSlipCentroid(faultxdmf, faultxyz, events=0):
```
and 
```
def ApproximateEventDurationAndHypocenter(faultxyz):
```
(to make the dependence of this function explicit).

Reply:
Ah ok, I got it now and fixed it